sycl/source/detail/scheduler/commands.cpp

```diff
@@ -245,6 +246,37 @@ cl_int MemCpyCommandHost::enqueueImp() {
   return CL_SUCCESS;
 }
 
+// SYCL has a parallel_for_work_group variant where the only NDRange
+// characteristics set by a user is the number of work groups. This does not map
+// to the OpenCL clEnqueueNDRangeAPI, which requires global work size to be set
+// as well. This function determines local work size based on the device
+// characteristics and the number of work groups requested by the user, then
+// calculates the global work size.
+static void adjustNDRangePerKernel(NDRDescT &NDR, cnri_kernel Kernel,
+                                   cnri_device Device) {
+  if (NDR.GlobalSize[0] != 0)
+    return; // GlobalSize is set - no need to adjust
+  // check the prerequisites:
+  assert(NDR.NumWorkGroups[0] != 0 && NDR.LocalSize[0] == 0);
+  range<3> WGSize = get_kernel_work_group_info_cl<
+      range<3>,
+      cl::sycl::info::kernel_work_group::compile_work_group_size>::_(Kernel,
```

Comment:
To improve performance we should cache this information in SYCL runtime instead of requesting it through lower level API, but this is a separate task. Could you add TODO here, please?

Reply:
ok.